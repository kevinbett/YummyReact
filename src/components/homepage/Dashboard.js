import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import Paper from 'material-ui/Paper';
import { orange700} from 'material-ui/styles/colors';
import '../../static/css/home.css';
import Navigation from '../homepage/Home';
import ChangePassword from '../user/Change_password';
import CategoryGet from '../categories/Categories';
import { Route } from "react-router-dom";
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import Recipes from '../recipes/Recipes';
import Notifications from 'react-notify-toast';
import {notify} from 'react-notify-toast';



const paperStyle = {
    height: '85%',
    width: "80%",
    marginLeft: '19%',
    textAlign: 'center',
    display: 'inline-block',
    backgroundColor: 'transparent',    
};

class Dashboard extends Component {
    constructor(props) {
        super(props);      
        this.state = {open: false, username:'', error:'', message:''}
        this.handleLogin = this.handleLogin.bind(this);
        this.handleLogout = this.handleLogout.bind(this);
        this.handleChangePassword = this.handleChangePassword.bind(this);
        this.componentWillMount = this.componentWillMount.bind(this);                          
    }
    handleLogin(event){
        event.preventDefault();
        this.props.history.push(`/login`);
    }
    handleLogout = () => {
        window.localStorage.removeItem('token');
        window.localStorage.setItem('logged_in', false)
        window.localStorage.setItem('message', "Successfully logged out")
        window.location.assign('/login');
        notify.show("Successfully logged out", 'success', 4000);                
    }
    handleChangePassword = (event) => {
        event.preventDefault();
        window.location.assign('/change-password');                
    }
    
    handleToggle = () => this.setState({open: !this.state.open});

    componentWillMount(){
        if (!window.localStorage.getItem('token')) {
            window.location.assign('/login');
        };
    }
    render() {
        if (!window.localStorage.getItem('token')) {
            window.location.assign('/login');
        };
        const rightButtons = (
            <div className="appBarIcons">
            <IconMenu
                iconButtonElement={<IconButton><MoreVertIcon /></IconButton>}
                anchorOrigin={{horizontal: 'left', vertical: 'bottom'}}
                targetOrigin={{horizontal: 'left', vertical: 'top'}}
                >
                <MenuItem primaryText="Change Password" onClick={this.handleChangePassword}/>
                <MenuItem primaryText="Sign out" onClick={this.handleLogout}/>
            </IconMenu>
            </div>
        );
        return (
            <MuiThemeProvider>
                <div className="landing">
                        <AppBar
                        onLeftIconButtonClick={this.handleToggle}
                        showMenuIconButton={false}
                        iconElementRight={rightButtons}
                        style={{backgroundColor: orange700}}>
                        <Notifications options={{zIndex: 5000}}/>
                        </AppBar>
                    <Paper style={paperStyle} zDepth={5}>                        
                        <div>                                                                                  
                            <Route exact path="/categories" component={CategoryGet} />
                            <Route exact path="/category/:id/recipes/" component={Recipes} />
                            <Route exact path="/change-password" component={ChangePassword} />                                                                             
                        </div>
                    </Paper>              
                    <Navigation/>
            </div>
            </MuiThemeProvider>
        );
    }  
}

export default Dashboard;