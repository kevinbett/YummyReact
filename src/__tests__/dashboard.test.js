import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Dashboard from '../homepage/dashboard';

describe('DashboardComponent', () => {
    const wrapper = shallow(<Dashboard />);
    it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});