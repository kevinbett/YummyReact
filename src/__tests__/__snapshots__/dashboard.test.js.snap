// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent renders without crashing 1`] = `
<MuiThemeProvider>
  <div
    className="landing"
  >
    <AppBar
      iconElementRight={
        <ToolbarGroup
          firstChild={false}
          lastChild={false}
        >
          <FlatButton
            disabled={false}
            fullWidth={false}
            label={null}
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
              }
            }
          />
          <IconMenu
            anchorOrigin={
              Object {
                "horizontal": "left",
                "vertical": "top",
              }
            }
            animated={true}
            clickCloseDelay={200}
            iconButtonElement={
              <IconButton
                disableTouchRipple={false}
                disabled={false}
                iconStyle={Object {}}
                tooltipPosition="bottom-center"
                touch={false}
              >
                <NavigationMoreVert />
              </IconButton>
            }
            multiple={false}
            onClick={[Function]}
            onItemClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onRequestChange={[Function]}
            open={null}
            targetOrigin={
              Object {
                "horizontal": "left",
                "vertical": "bottom",
              }
            }
            useLayerForClickAway={false}
          >
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Change password"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
            />
            <MenuItem
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              onClick={[Function]}
              primaryText="Sign out"
              targetOrigin={
                Object {
                  "horizontal": "left",
                  "vertical": "top",
                }
              }
            />
          </IconMenu>
        </ToolbarGroup>
      }
      onLeftIconButtonClick={[Function]}
      showMenuIconButton={false}
      style={
        Object {
          "backgroundColor": "#f57c00",
        }
      }
      title=""
      zDepth={1}
    >
      <_class
        options={
          Object {
            "zIndex": 5000,
          }
        }
      />
    </AppBar>
    <LeftDrawer />
    <Paper
      circle={false}
      rounded={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "display": "inline-block",
          "height": "85%",
          "marginLeft": "19%",
          "textAlign": "center",
          "width": "80%",
        }
      }
      transitionEnabled={true}
      zDepth={5}
    >
      <div>
        <Route
          component={[Function]}
          exact={true}
          path="/dashboard/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/dashboard/categories"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/dashboard/category/:id/recipes/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/dashboard/category"
        />
      </div>
    </Paper>
    <Navigation />
  </div>
</MuiThemeProvider>
`;
